src/supervisor-api.ts

```diff
@@ -208,17 +227,28 @@ export class SupervisorAPI {
 				log.debug('Supervisor API listening on allowed interfaces only');
 			}
 		} catch (err) {
-			log.error(
-				'Error on switching supervisor API listening rules - stopping API.\n',
-				err,
-			);
-			this.stop();
+			log.error('Error on switching supervisor API listening rules', err);
+			return this.stop({ errored: true });
 		}
 	}
 
-	public stop() {
+	public async stop(options?: SupervisorAPIStopOpts): Promise<void> {
```

Comment:
The stop function offers a callbacks so I return a promise that allows the caller to know when this async operation is fully complete. I also am returning the error so that the person calling it knows how to handle the unsuccessful close.